astropy/coordinates/sky_coordinate.py

```diff
@@ -364,6 +364,8 @@ def __eq__(self, value):
         equivalent, extra frame attributes are equivalent, and that the
         representation data are exactly equal.
         """
+        if not isinstance(value, SkyCoord):
+            return False
```

Comment:
The pythonic route is to `return NotImplemented` - this means `value` gets a chance as well to see if it knows how to compare. If a string, it will also return `NotImplemented`, and then python returns `False`. For instance,
```
In [11]: class AlwaysEqual:
    ...:     def __eq__(self, other):
    ...:         return True
    ...: 
    ...: 

In [12]: always_equal = AlwaysEqual()

In [13]: 'abc'.__eq__(always_equal)
Out[13]: NotImplemented

In [14]: 'abc' == always_equal
Out[14]: True

In [15]: obj = object()

In [16]: 'abc'.__eq__(obj)
Out[16]: NotImplemented

In [17]: obj.__eq__('abc')
Out[17]: NotImplemented
```

And with `SkyCoord` before your PR:
```
In [22]: always_equal == sc
Out[22]: True

In [23]: sc = SkyCoord([0, 10]*u.deg, [0, 10]*u.deg)

In [24]: always_equal == sc
Out[24]: True

In [25]: sc == always_equal
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/work/scintillometry/psrb1929_fast/python/wavefield.py in <module>
----> 1 sc == always_equal

/usr/lib/python3/dist-packages/astropy/coordinates/sky_coordinate.py in __eq__(self, value)
    366         """
    367         # Make sure that any extra frame attribute names are equivalent.
--> 368         for attr in self._extra_frameattr_names | value._extra_frameattr_names:
    369             if not self.frame._frameattr_equiv(getattr(self, attr),
    370                                                getattr(value, attr)):

AttributeError: 'AlwaysEqual' object has no attribute '_extra_frameattr_names'
```
With your PR as is, the second one would return `False`, even though it should return `True`.

Reply:
Thanks for the explanation!